cadc-util/src/main/java/ca/nrc/cadc/net/HttpPost.java

```diff
@@ -425,30 +426,33 @@ private void doPost(HttpURLConnection conn, Map<String,List<Object>> params, Map
         
         log.debug("params: " + sb.toString());
         
-        OutputStreamWriter writer = new OutputStreamWriter(conn.getOutputStream(), "UTF-8");
-        writer.write(sb.toString());
-        
-        if (multi)
-        {
-            for (Map.Entry<String,Object> up : uploads.entrySet())
+        Charset utf8 = Charset.forName("UTF-8");
+        OutputStream writer = conn.getOutputStream();
+        try {
+            writer.write(sb.toString().getBytes(utf8));
+
+            if (multi)
             {
-                if (up.getValue() instanceof File) {
-                    writeFilePart(up.getKey(), ((File)up.getValue()), writer);
-                }
-                else if (up.getValue() instanceof FileContent) {
-                    writeFilePart(up.getKey(), (FileContent)up.getValue(), writer);
-                }
-                else {
-                    throw new UnsupportedOperationException("Unexpected upload type: " + up.getClass().getName());
+                for (Map.Entry<String,Object> up : uploads.entrySet())
+                {
```

Comment:
May as well update the codestyle here so that open brackets are on the same line as the condition.

Reply:
I have been reluctant to do that and hide  real changes. Maybe delay that and have a with just checkstyle fix?